js/search/controllers/search.controller.js

```diff
@@ -24,14 +24,42 @@
      * @ngdoc controller
      * @name jwShowcase.search.SearchController
      */
-    SearchController.$inject = ['$state', 'platform', 'searchFeed'];
-    function SearchController ($state, platform, searchFeed) {
+    SearchController.$inject = ['$scope', '$state', '$stateParams', 'platform', 'searchFeed', 'api'];
+    function SearchController ($scope, $state, $stateParams, platform, searchFeed, api) {
 
         var vm = this;
 
-        vm.feed = searchFeed;
+        vm.cardClickHandler     = cardClickHandler;
+        vm.showMoreClickHandler = showMoreClickHandler;
+        vm.itemsLeft            = itemsLeft;
 
-        vm.cardClickHandler = cardClickHandler;
+        vm.feed = null;
+
+        var limit = 10;
+
+        if($stateParams.searchInCaptions) {
+            var partFeed = angular.copy(searchFeed);
+
+            partFeed.playlist = searchFeed.playlist.slice(0, limit);
+
+            addItemsToFeed(partFeed.playlist).then(function () {
+                vm.feed = partFeed;
+
+                $scope.$apply();
+            });
+        } else {
+            vm.feed = searchFeed;
+        }
+
+        function addItemsToFeed(items) {
+            var query = $stateParams.query.replace(/\+/g, ' ');
+
+            var patchCaptionPromises = items.map(function (item) {
+                return api.patchCaptions(item, query);
```

Comment:
Wrong name. You're not patching captions, you're patching items with captions.

Reply:
Your own function is called patchCaption. So all the promises are then called patchCaptionPromises.

Reply:
@brniamut It's software. It can be changed. I didn't assign blame to you, I said it needs to be renamed.

But if it helps you, yes. I chose a bad name.

Reply:
Didn't mean to blame you at all. Just tried to explain why I made the choice to call it like that. Agree with the name change.